client/src/components/Forms/CourseTaskSelect.tsx

```diff
@@ -1,20 +1,50 @@
 import * as React from 'react';
 import { Select, Form } from 'antd';
+import _ from 'lodash';
+
 import { CourseTask } from 'services/course';
 
-type Props = { data: CourseTask[]; onChange?: (id: number) => void };
+type Props = {
+  data: CourseTask[];
+  onChange?: (id: number) => void;
+};
+
+const MAX_DATE = '+275760-09-13T00:00:00.000Z';
 
 export function CourseTaskSelect(props: Props) {
-  const { data, onChange, ...rest } = props;
-  const selectProps = onChange ? { onChange } : {};
+  const { data, onChange, ...options } = props;
+  data.sort(function (firstTask, secondTask) {
+    return Date.parse(secondTask.studentEndDate ?? MAX_DATE) - Date.parse(firstTask.studentEndDate ?? MAX_DATE);
+  });
+  const selectProps = _.omitBy({ onChange }, _.isNil);
```

Comment:
i wonder why do we need `lodash` here?

Reply:
to remove null/undefined value from an object

Reply:
Previously it was:
```
const selectProps = onChange ? { onChange } : {};
```

Now component requires `lodash`, has more code but does the same.
Please revert it unless you have strong reason to use third-party library here.

Reply:
Done